Week3/BusinessCard/app/src/main/java/com/raywenderlich/businesscard/nivaldolopo/MainActivity.kt

```diff
@@ -0,0 +1,53 @@
+package com.raywenderlich.businesscard.nivaldolopo
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.animation.AnimationUtils
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+
+class MainActivity : AppCompatActivity() {
+
+    internal lateinit var buttonQRCode: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        buttonQRCode = findViewById(R.id.buttonQRCode)
+
+        buttonQRCode.setOnClickListener { view ->
+            //set animation to the button
+            val bounceAnimation = AnimationUtils.loadAnimation(this, R.anim.bounce)

```

Comment:
I'd separate this into anotehr small function, as well as the startActivity code!